### PR TITLE
add Upload Dumps on Crash option

### PR DESCRIPTION
**Pull request by IrinaShkviro**
Added `Upload Dumps on Crash` to drop-down Options menu. This setting is saved as m_UploadDumpsToServer in config.xml. Default value is false.

Moved SetUploadsEnabled call from constructor to the separate function that called a bit later (after config.xml is read when it's possible to read options). In this case all dumps that created between these 2 events (CrashHandler created and config.xml read) will be written only in the database and not send to server.

### CODE REVIEW

File: OrbitCore/CoreApp.h [hunk 1]
```diff
@@ -37,6 +38,7 @@ class CoreApp {
   virtual bool GetUnsafeHookingEnabled() { return false; }
   virtual bool GetSamplingEnabled() { return false; }
   virtual bool GetOutputDebugStringEnabled() { return false; }
+  virtual bool GetUploadDumpsToServerEnabled() { return false; }
```
Comment (beckerhe) [inline OrbitCore/CoreApp.h:41]: Make it const, even though the other functions aren't (but should be!).

File: OrbitCore/CrashHandler.h [hunk 2]
```diff
@@ -12,13 +14,14 @@ class CrashHandler {
  public:
   explicit CrashHandler(const std::string& dump_path,
                         const std::string& handler_path,
-                        const std::string& crash_server_url,
-                        bool is_upload_enabled);
+                        const std::string& crash_server_url);
   void DumpWithoutCrash() const;
+  void SetUploadsEnabled(bool is_upload_enabled) const;
```
Comment (beckerhe) [inline OrbitCore/CrashHandler.h:19]: Don't make this const even though it's technically correct. The better solution would be to not put crash_report_db_ in a unique_ptr but sadly that's not possible due to bad API.

File: OrbitCore/CrashHandler.cpp [hunk 3]
```diff
@@ -64,6 +57,15 @@ CrashHandler::CrashHandler(const std::string& dump_path,
                                 /*asynchronous_start=*/false);
 }
 
+//-----------------------------------------------------------------------------
+void CrashHandler::SetUploadsEnabled(bool is_upload_enabled) const {
+  // set user preferences for dumps submission to collection server
+  if (crash_report_db_ != nullptr &&
+      crash_report_db_->GetSettings() != nullptr) {
+    crash_report_db_->GetSettings()->SetUploadsEnabled(is_upload_enabled);
```
Comment (beckerhe) [inline OrbitCore/CrashHandler.cpp:65]: You're checking for nullptr and then you call GetSettings again which could in theory give you a different result.
Better store the settings pointer in a variable!

**Comment by IrinaShkviro** [comment]
> Small comments and one question: When the user changes the setting: I don't see how this calls crash_report_db->GetSettings()->SetUploadsEnabled.

Yes, you're right, it doesn't change this setting immediately. I assumed that the most common usage would be to set it once and then just read from file on start, so I just added it as all other options. If we want to change it immediately I also should add saving the config.xml on action triggered.

**Comment by beckerhe** [comment]
> > Small comments and one question: When the user changes the setting: I don't see how this calls crash_report_db->GetSettings()->SetUploadsEnabled.
> 
> Yes, you're right, it doesn't change this setting immediately. I assumed that the most common usage would be to set it once and then just read from file on start, so I just added it as all other options. If we want to change it immediately I also should add saving the config.xml on action triggered.

Sounds good. For reference, that was my train of thought: I'm a user and I decide to disable uploading when it previously was enabled. So I click the action, then Orbit crashes and this crash dump gets uploaded and my settings change does not get persisted. So this should/may not happen.

**Comment by IrinaShkviro** [comment]
> > > Small comments and one question: When the user changes the setting: I don't see how this calls crash_report_db->GetSettings()->SetUploadsEnabled.
> > 
> > 
> > Yes, you're right, it doesn't change this setting immediately. I assumed that the most common usage would be to set it once and then just read from file on start, so I just added it as all other options. If we want to change it immediately I also should add saving the config.xml on action triggered.
> 
> Sounds good. For reference, that was my train of thought: I'm a user and I decide to disable uploading when it previously was enabled. So I click the action, then Orbit crashes and this crash dump gets uploaded and my settings change does not get persisted. So this should/may not happen.

Agree with you, in case when we change setting and then Orbit crashed it looks weird. I'll convert the PR to draft as it will take some time to implement it.

File: OrbitGl/ApplicationOptions.h [hunk 4]
```diff
@@ -15,6 +17,8 @@ struct ApplicationOptions {
   std::string asio_server_address;
   // GRPC connection string
   std::string grpc_server_address;
+  
+  CrashHandler* crash_handler;
```
Comment (beckerhe) [inline OrbitGl/ApplicationOptions.h:21]: Can you initialize that with a nullptr?

File: OrbitGl/App.cpp [hunk 5]
```diff
@@ -961,6 +963,19 @@ void OrbitApp::OnProcessSelected(uint32_t pid) {
...
 }
 
+//-----------------------------------------------------------------------------
+bool OrbitApp::GetUploadDumpsToServerEnabled() const {
+  return GParams.m_UploadDumpsToServer;
+}
+
+//-----------------------------------------------------------------------------
+void OrbitApp::EnableUploadDumpsToServer(bool a_Value) {
+  options_.crash_handler->SetUploadsEnabled(a_Value);
```
Comment (beckerhe) [inline OrbitGl/App.cpp:973]: Can you check whether options_.crash_handler is not a nullptr before calling SetUploadsEnabled?

**Comment by IrinaShkviro** [comment]
> LGTM. Thank you. Two small comments.

Updated PR. Thanks for the review!

**Pull request merged**